client/src/backend/containers/Records/export-targets/Edit.js

```diff
@@ -0,0 +1,110 @@
+import React, { PureComponent } from "react";
+import PropTypes from "prop-types";
+import connectAndFetch from "utils/connectAndFetch";
+import withConfirmation from "hoc/with-confirmation";
+import { entityStoreActions } from "actions";
+import { select } from "utils/entityUtils";
+import { exportTargetsAPI, requests } from "api";
+import lh from "helpers/linkHandler";
+import Navigation from "backend/components/navigation";
+import Form from "./Form";
+
+const { request, flush } = entityStoreActions;
+
+export class ExportTargetsEditContainer extends PureComponent {
+  static mapStateToProps = state => {
+    return {
+      exportTarget: select(requests.beExportTarget, state.entityStore)
+    };
+  };
+
+  static displayName = "ExportTargets.Edit";
+
+  static propTypes = {
+    // export target information is required, but can be null/undefined until
+    // the async network request is fulfilled
+    exportTarget: PropTypes.object, // eslint-disable-line react/require-default-props
```

Comment:
I know that the eslint rule `react/require-default-props` is turned off for this project, but I think this is a helpful bit of information for any developer that is not familiar with the way manifold uses `mapStateToProps`. When trying to build this component, I did get tripped up that the value of exportTarget would start out as `null`, and then fill in the data after the network request was fulfilled.

Reply:
Having this one-off rule disable here doesn't help. I'm sorry you found this confusing, but it hasn't been a problem for other devs on the process. Null is exactly the state that we want the prop to be in, because null tells us that there is no data there. If we wanted to change this behavior/convention, we'd be changing it at a much higher level, not as a one-off on this one component.